File: backend/voting-app/src/main/java/com/tu/votingapp/entities/elections/ElectionEntity.java
package com.tu.votingapp.entities.elections;

import com.tu.votingapp.entities.UserEntity;
import com.tu.votingapp.enums.ElectionStatus;
import com.tu.votingapp.enums.ElectionType;
import jakarta.persistence.*;
import lombok.*;

import java.sql.Date;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "elections")
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class ElectionEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @Column(name = "election_name")
    private String electionName;

    @Column(name = "election_description")
    private String description;

    @Column(name = "start_date")
    private Date startDate;

    @Column(name = "end_date")
    private Date endDate;

    @Column(name = "election_type")
    private ElectionType electionType;

    @Enumerated(EnumType.STRING)
    @Column(name = "election_status")
    private ElectionStatus status;

    @OneToMany(mappedBy = "election", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<CandidateEntity> candidates = new ArrayList<>();

    @OneToMany(mappedBy = "election", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PartyEntity> parties = new ArrayList<>();


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private UserEntity createdBy;
}

File: frontend/android/app/src/main/java/com/example/android/entity/election/ElectionEntity.kt
package com.example.android.entity.election

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.android.ui.theme.viewmodels.ElectionDisplayItem

@Entity(tableName = "elections")
data class ElectionEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long = 0,

    @ColumnInfo(name = "election_name")
    val electionName: String,

    @ColumnInfo(name = "description")
    val description: String? = null,

    @ColumnInfo(name = "start_date")
    val startDate: String, // Use ISO date format (e.g. 2024-05-01)

    @ColumnInfo(name = "end_date")
    val endDate: String,

    @ColumnInfo(name = "election_type")
    val electionType: String, // Store enum as String (e.g. "PRESIDENTIAL")

    @ColumnInfo(name = "status")
    val status: String, // Store enum as String (e.g. "ONGOING")

    @ColumnInfo(name = "created_by")
    val createdById: Long
) {
    fun toDisplayItem(): ElectionDisplayItem {
        return ElectionDisplayItem(
            id = this.id,
            name = this.electionName
        )
    }
}